internal/api/handlers.go

```diff
@@ -25,6 +29,10 @@ func NewGoCertRouter(env *Environment) http.Handler {
 	v1 := http.NewServeMux()
 	v1.HandleFunc("GET /status", HealthCheck)
 	v1.Handle("/api/v1/", http.StripPrefix("/api/v1", router))
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(collectors.NewGoCollector())
+	prometheusHandler := promhttp.HandlerFor(reg, promhttp.HandlerOpts{})
+	v1.Handle("/api/v1/metrics", prometheusHandler)
```

Comment:
I'd recommend treating metrics as their own independent web server and not placing them under `internal/api/handlers.go`. Instead, they can be under `/metrics/metrics.go` and directly started from `main` via something like `go metrics.Init()`.

## Rationale
1. The metrics server is independent from the other gocert specific endpoints and can be maintain on its own. It does not make sense for it to be under `api/v1` in my view, it's not part of the gocert api.

2. If for whatever reason the technology we use for metrics changes from Prometheus to something else, all we need to change is `internal/metrics`, just like we do for the db.

Reply:
While it may make sense to have a separate server for collecting metrics for the go application, it will severely limit the information we can get from the GoCert server. Unless we manually watch stdout, stderr and the db, we can not collect any metrics regarding the HTTP requests or responses, or what action the server has taken, only what it can infer from external data.

I do agree that it should not be under `api/v1`, and agree that this implementation is too coupled to prometheus.

I recommend writing a `Metrics` function that returns an object that implements the http.Handler interface, which can be swapped out for any other metric provider while keeping the same level of access to the GoCert API server.

Reply:
I created a metrics package, there an abstract metrics handler and a prometheus metrics handler.
I renamed the Mux from v1.

Let me know what you think. @gruyaume @Kayra1